### PR TITLE
Enable thermostat init function

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Thermostat init  function not used/enabled.

#### Change overview
Enable thermostat init callback function.
Fix signatures
Sort enabling list
Use attribute store to keep featuremap value (but unsure if this is correct or works)
Fix code compilation

#### Testing

Not tested, hence draft.

### CODE REVIEW

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 1]
```diff
@@ -557,13 +558,9 @@ bool emberAfThermostatClusterSetpointRaiseLowerCallback(app::CommandHandler * co
...
     uint32_t FeatureMap                      = 0;
 
-    // TODO re-enable reading reaturemap once https://github.com/project-chip/connectedhomeip/pull/9725 is merged
-#ifndef FEATURE_MAP_OVERIDE
-    emberAfReadServerAttribute(endpoint, Thermostat::Id, chip::app::Clusters::Globals::Attributes::Ids::FeatureMap,
-                               (uint8_t *) &FeatureMap, sizeof(FeatureMap));
-#else
-    FeatureMap = FEATURE_MAP_OVERIDE;
-#endif
+    emberAfReadServerAttribute(aEndpointId, chip::app::Clusters::Thermostat::Id,
```
Comment (andy31415) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:561]: @bzbarsky-apple - I would like to use accessors here but do not know how. Help  would be appreciated.

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 2]
```diff
@@ -83,6 +81,13 @@ void emberAfThermostatClusterServerInitCallback()
     // with weak binding so that real thermostat
     // can get the values.
     // or should this just be the responsibility of the thermostat application?
+
+    // Feature map override, for testing only:
+    uint32_t featureMap = FEATURE_MAP_HEAT | FEATURE_MAP_COOL | FEATURE_MAP_AUTO;
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:86]: Why are we overriding the value set in the actual app configuration (in ZAP) by default?  That does not seem right...  I know the current code is doing it, but a PR just landed to stop that, yes?

File: src/app/clusters/thermostat-server/thermostat-server.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/clusters/thermostat-server/thermostat-server.cpp:561]: Need to fix accessors. https://github.com/project-chip/connectedhomeip/pull/12808

File: src/app/clusters/thermostat-server/thermostat-server.cpp (hunk 2 above)
Comment (chrisbeach59) [reply src/app/clusters/thermostat-server/thermostat-server.cpp:86]: This remains only for testing purposes.  Rebase and regen process to test code changes that are dependent on Feature Map contents are very time consuming.

File: src/app/clusters/thermostat-server/thermostat-server.cpp (hunk 1 above)
Comment (chrisbeach59) [reply src/app/clusters/thermostat-server/thermostat-server.cpp:561]: @bzbarsky-apple @andy31415 Thanks.  I will  rebase this branch, and test this change shortly, then promote to non draft.

**Comment by chrisbeach59** [comment]
Suggest to Kill this PR.  
I made the helper.js and rebuilt and made the  desired thermostat functionality in PR 12863. 
Made sure that Andrei and Boris  are both on as approvers for review

**Pull request closed without merging**